packages/js/product-editor/src/blocks/product-fields/variation-options/edit.tsx

```diff
@@ -57,19 +58,26 @@ export function Edit( {
 			'default_attributes'
 		);
 
-	const { postType } = context;
 	const productId = useEntityId( 'postType', postType );
 
-	const { attributes, handleChange } = useProductAttributes( {
-		allAttributes: entityAttributes,
-		isVariationAttributes: true,
-		productId: useEntityId( 'postType', 'product' ),
-		onChange( values, defaultAttributes ) {
-			setEntityAttributes( values );
-			setEntityDefaultAttributes( defaultAttributes );
-			generateProductVariations( values, defaultAttributes );
-		},
-	} );
+	const { attributes, fetchAttributes, handleChange } = useProductAttributes(
+		{
+			allAttributes: entityAttributes,
+			isVariationAttributes: true,
+			productId,
+			onChange( values, defaultAttributes ) {
+				setEntityAttributes( values );
+				setEntityDefaultAttributes( defaultAttributes );
+				generateProductVariations( values, defaultAttributes );
+			},
+		}
+	);
+
+	useEffect( () => {
+		if ( isInSelectedTab ) {
+			fetchAttributes();
+		}
+	}, [ isInSelectedTab, entityAttributes ] );
```

Comment:
I think we should also relay on `isInSelectedTab` within the attributes block that is rendered as part of the `Organization` tab since the term request is being executed still when the product has attributes.

Reply:
Done in [df61c4c](https://github.com/woocommerce/woocommerce/pull/47361/commits/df61c4cfdd9d0aa60be3f6fb4aa7e9f042f61237)